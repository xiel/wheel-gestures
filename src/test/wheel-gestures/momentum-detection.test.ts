import doubleSwipeRight from '../fixtures/double-swipe-right.json'
import swipeDownFast from '../fixtures/swipe-down-fast-trackpad.json'
import swipeDown from '../fixtures/swipe-down-trackpad.json'
import swipeLeftFast from '../fixtures/swipe-left-fast-trackpad.json'
import swipeLeftMouseFF from '../fixtures/swipe-left-mouse-FF.json'
import swipeLeft from '../fixtures/swipe-left-trackpad.json'
import swipeRight from '../fixtures/swipe-right.json'
import swipeRightFast from '../fixtures/swipe-right-fast.json'
import swipeUpFast from '../fixtures/swipe-up-fast-trackpad.json'
import swipeUp from '../fixtures/swipe-up-trackpad.json'
import { recordPhases } from '../helper/recordPhases'

describe('momentum detection recognizes', () => {
  it('swipe right', () => {
    expect(recordPhases(swipeRight.wheelEvents)).toMatchSnapshot()
  })

  it('swipe right - fast', () => {
    expect(recordPhases(swipeRightFast.wheelEvents)).toMatchSnapshot()
  })

  it('swipe left', () => {
    expect(recordPhases(swipeLeft.wheelEvents)).toMatchSnapshot()
  })

  it('swipe left (mouse) - Firefox', () => {
    expect(recordPhases(swipeLeftMouseFF.wheelEvents)).toMatchSnapshot()
  })

  it('swipe left - fast', () => {
    expect(recordPhases(swipeLeftFast.wheelEvents)).toMatchSnapshot()
  })

  it('swipe up', () => {
    expect(recordPhases(swipeUp.wheelEvents)).toMatchSnapshot()
  })

  it('swipe up - fast', () => {
    expect(recordPhases(swipeUpFast.wheelEvents)).toMatchSnapshot()
  })

  it('swipe down', () => {
    expect(recordPhases(swipeDown.wheelEvents)).toMatchSnapshot()
  })

  it('swipe down - fast', () => {
    expect(recordPhases(swipeDownFast.wheelEvents)).toMatchSnapshot()
  })

  it('double swipe right', () => {
    expect(recordPhases(doubleSwipeRight.wheelEvents)).toMatchSnapshot()
  })
})
