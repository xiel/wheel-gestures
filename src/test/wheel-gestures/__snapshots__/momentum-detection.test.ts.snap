// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`momentum detection recognizes double swipe right 1`] = `
Object {
  "phases": Array [
    Object {
      "axisDelta": Array [
        0,
        0,
        0,
      ],
      "axisMovement": Array [
        -952,
        8,
        0,
      ],
      "axisVelocity": Array [
        -0.682340725381047,
        0,
        0,
      ],
      "event": Object {
        "deltaMode": 0,
        "deltaX": -2,
        "deltaY": 0,
        "timeStamp": 3317.0500000123866,
      },
      "hasPrevious": true,
      "isEnding": true,
      "isMomentum": true,
      "isMomentumCancel": true,
      "isStart": false,
      "range": Array [
        32,
        33,
      ],
      "wheelType": "momentum",
    },
    Object {
      "axisDelta": Array [
        0,
        0,
        0,
      ],
      "axisMovement": Array [
        -1755,
        -2,
        0,
      ],
      "axisVelocity": Array [
        -0.039928129363731614,
        0,
        0,
      ],
      "event": Object {
        "deltaMode": 0,
        "deltaX": -1,
        "deltaY": 0,
        "timeStamp": 4477.8150000202,
      },
      "hasPrevious": true,
      "isEnding": true,
      "isMomentum": true,
      "isMomentumCancel": false,
      "isStart": false,
      "range": Array [
        102,
        102,
      ],
      "wheelType": "momentum",
    },
  ],
}
`;

exports[`momentum detection recognizes swipe down - fast 1`] = `
Object {
  "phases": Array [
    Object {
      "axisDelta": Array [
        0,
        0,
        0,
      ],
      "axisMovement": Array [
        -23,
        -7290,
        0,
      ],
      "axisVelocity": Array [
        0,
        -0.04081632653061224,
        0,
      ],
      "event": Object {
        "deltaMode": 0,
        "deltaX": 0,
        "deltaY": -1,
        "timeStamp": 30302,
      },
      "hasPrevious": true,
      "isEnding": true,
      "isMomentum": true,
      "isMomentumCancel": false,
      "isStart": false,
      "range": Array [
        95,
        95,
      ],
      "wheelType": "momentum",
    },
  ],
}
`;

exports[`momentum detection recognizes swipe down 1`] = `
Object {
  "phases": Array [
    Object {
      "axisDelta": Array [
        0,
        0,
        0,
      ],
      "axisMovement": Array [
        7,
        -2134,
        0,
      ],
      "axisVelocity": Array [
        0,
        -0.04878048780487805,
        0,
      ],
      "event": Object {
        "deltaMode": 0,
        "deltaX": 0,
        "deltaY": -1,
        "timeStamp": 7709.000000000001,
      },
      "hasPrevious": true,
      "isEnding": true,
      "isMomentum": true,
      "isMomentumCancel": false,
      "isStart": false,
      "range": Array [
        87,
        87,
      ],
      "wheelType": "momentum",
    },
  ],
}
`;

exports[`momentum detection recognizes swipe left (mouse) - Firefox 1`] = `
Object {
  "phases": Array [
    Object {
      "axisDelta": Array [
        0,
        0,
        0,
      ],
      "axisMovement": Array [
        740,
        0,
        0,
      ],
      "axisVelocity": Array [
        0.04,
        0,
        0,
      ],
      "event": Object {
        "deltaMode": 0,
        "deltaX": 1,
        "deltaY": 0,
        "timeStamp": 22026,
      },
      "hasPrevious": true,
      "isEnding": true,
      "isMomentum": true,
      "isMomentumCancel": false,
      "isStart": false,
      "range": Array [
        54,
        54,
      ],
      "wheelType": "momentum",
    },
  ],
}
`;

exports[`momentum detection recognizes swipe left - fast 1`] = `
Object {
  "phases": Array [
    Object {
      "axisDelta": Array [
        0,
        0,
        0,
      ],
      "axisMovement": Array [
        9964,
        -8,
        0,
      ],
      "axisVelocity": Array [
        0.039603960396039604,
        0,
        0,
      ],
      "event": Object {
        "deltaMode": 0,
        "deltaX": 1,
        "deltaY": 0,
        "timeStamp": 147746,
      },
      "hasPrevious": true,
      "isEnding": true,
      "isMomentum": true,
      "isMomentumCancel": false,
      "isStart": false,
      "range": Array [
        100,
        100,
      ],
      "wheelType": "momentum",
    },
  ],
}
`;

exports[`momentum detection recognizes swipe left 1`] = `
Object {
  "phases": Array [
    Object {
      "axisDelta": Array [
        0,
        0,
        0,
      ],
      "axisMovement": Array [
        3001,
        -15,
        0,
      ],
      "axisVelocity": Array [
        0.04,
        0,
        0,
      ],
      "event": Object {
        "deltaMode": 0,
        "deltaX": 1,
        "deltaY": 0,
        "timeStamp": 2925.0000000000005,
      },
      "hasPrevious": true,
      "isEnding": true,
      "isMomentum": true,
      "isMomentumCancel": false,
      "isStart": false,
      "range": Array [
        82,
        82,
      ],
      "wheelType": "momentum",
    },
  ],
}
`;

exports[`momentum detection recognizes swipe right - fast 1`] = `
Object {
  "phases": Array [
    Object {
      "axisDelta": Array [
        0,
        0,
        0,
      ],
      "axisMovement": Array [
        -15,
        0,
        0,
      ],
      "axisVelocity": Array [
        -0.03963731853433476,
        0,
        0,
      ],
      "event": Object {
        "deltaMode": 0,
        "deltaX": -1,
        "deltaY": 0,
        "timeStamp": 64035.86500001256,
      },
      "hasPrevious": true,
      "isEnding": true,
      "isMomentum": false,
      "isMomentumCancel": false,
      "isStart": false,
      "range": Array [
        0,
        9,
      ],
      "wheelType": "user",
    },
    Object {
      "axisDelta": Array [
        0,
        0,
        0,
      ],
      "axisMovement": Array [
        -12774,
        0,
        0,
      ],
      "axisVelocity": Array [
        -0.0400060009027491,
        0,
        0,
      ],
      "event": Object {
        "deltaMode": 0,
        "deltaX": -1,
        "deltaY": 0,
        "timeStamp": 65929.5800000109,
      },
      "hasPrevious": true,
      "isEnding": true,
      "isMomentum": true,
      "isMomentumCancel": false,
      "isStart": false,
      "range": Array [
        96,
        96,
      ],
      "wheelType": "momentum",
    },
  ],
}
`;

exports[`momentum detection recognizes swipe right 1`] = `
Object {
  "phases": Array [
    Object {
      "axisDelta": Array [
        0,
        0,
        0,
      ],
      "axisMovement": Array [
        -1634,
        0,
        0,
      ],
      "axisVelocity": Array [
        -0.0398585023158581,
        0,
        0,
      ],
      "event": Object {
        "deltaMode": 0,
        "deltaX": -1,
        "deltaY": 0,
        "timeStamp": 2458.8000000076136,
      },
      "hasPrevious": true,
      "isEnding": true,
      "isMomentum": true,
      "isMomentumCancel": false,
      "isStart": false,
      "range": Array [
        63,
        63,
      ],
      "wheelType": "momentum",
    },
  ],
}
`;

exports[`momentum detection recognizes swipe up - fast 1`] = `
Object {
  "phases": Array [
    Object {
      "axisDelta": Array [
        0,
        0,
        0,
      ],
      "axisMovement": Array [
        -38,
        8065,
        0,
      ],
      "axisVelocity": Array [
        0,
        0.0392156862745098,
        0,
      ],
      "event": Object {
        "deltaMode": 0,
        "deltaX": 0,
        "deltaY": 1,
        "timeStamp": 103107,
      },
      "hasPrevious": true,
      "isEnding": true,
      "isMomentum": true,
      "isMomentumCancel": false,
      "isStart": false,
      "range": Array [
        98,
        98,
      ],
      "wheelType": "momentum",
    },
  ],
}
`;

exports[`momentum detection recognizes swipe up 1`] = `
Object {
  "phases": Array [
    Object {
      "axisDelta": Array [
        0,
        0,
        0,
      ],
      "axisMovement": Array [
        -70,
        3836,
        0,
      ],
      "axisVelocity": Array [
        0,
        0.04,
        0,
      ],
      "event": Object {
        "deltaMode": 0,
        "deltaX": 0,
        "deltaY": 1,
        "timeStamp": 87480,
      },
      "hasPrevious": true,
      "isEnding": true,
      "isMomentum": true,
      "isMomentumCancel": false,
      "isStart": false,
      "range": Array [
        84,
        84,
      ],
      "wheelType": "momentum",
    },
  ],
}
`;
